pkg/operators/machine-api-operator.go

```diff
@@ -164,3 +167,71 @@ var _ = Describe("[Feature:Operators] Machine API cluster operator status should
 		Expect(framework.IsStatusAvailable(client, "machine-api")).To(BeTrue())
 	})
 })
+
+var _ = Describe("[Feature:Operators] When cluster-wide proxy is configured, Machine API cluster operator should ", func() {
+	var machineSet *mapiv1beta1.MachineSet
+	var proxyMachineSet *mapiv1beta1.MachineSet
+	var machineSetParams framework.MachineSetParams
+	It("reflect the configured proxy in machine-api-controller", func() {
+		By("creating host for proxy")
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+		proxyMachineSet, err = framework.CreateProxyMachineSet(client)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("deploying an HTTP proxy")
+		deployment, err := framework.DeployClusterProxy(client)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(framework.IsDeploymentAvailable(client, deployment.Name, deployment.Namespace)).To(BeTrue())
+
+		By("configuring cluster-wide proxy")
+		pods, err := framework.GetPods(client, map[string]string{"app": "squid"})
+		Expect(err).NotTo(HaveOccurred())
+		Expect(len(pods.Items) > 0).To(BeTrue())
+		proxy := configv1.Proxy{}
+		proxy.Spec.HTTPProxy = "http://" + pods.Items[0].Status.HostIP + ":80"
+		proxy.Spec.HTTPSProxy = "http://" + pods.Items[0].Status.HostIP + ":80"
+		framework.SetClusterProxy(client, proxy)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("waiting for machine-api-controller deployment to reflect configured cluster-wide proxy")
+		Expect(framework.WaitForProxyInjectionSync(client, maoManagedDeployment, framework.MachineAPINamespace, true)).To(BeTrue())
+	})
+
+	It("allow a machineset to be created and destroyed", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a machineset")
+
+		machineSetParams = framework.BuildMachineSetParams(client, 3)
+		machineSet, err = framework.CreateMachineSet(client, machineSetParams)
+		Expect(err).ToNot(HaveOccurred())
+		framework.WaitForMachineSet(client, machineSet.GetName())
+
+		By("destroying a machineset")
+		err = client.Delete(context.Background(), machineSet)
+		Expect(err).ToNot(HaveOccurred())
+		framework.WaitForMachineSetDelete(client, machineSet)
+	})
+
+	It("reflect an unconfigured proxy in machine-api-controller", func() {
+		client, err := framework.LoadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		By("unconfiguring cluster-wide proxy")
+		proxy := configv1.Proxy{}
+		proxy.Spec.HTTPProxy = ""
+		proxy.Spec.HTTPSProxy = ""
+		framework.SetClusterProxy(client, proxy)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("waiting for machine-api-controller deployment to reflect unconfigured cluster-wide proxy")
+		Expect(framework.WaitForProxyInjectionSync(client, maoManagedDeployment, framework.MachineAPINamespace, false)).To(BeTrue())
+		framework.DestroyClusterProxy(client)
+
+		Expect(proxyMachineSet != nil).To(BeTrue())
+		framework.DeleteMachineSets(client, proxyMachineSet)
+	})
```

Comment:
This either needs to be one `It` statement as below, or it needs to be a `BeforeEach` which sets up the cluster-wide proxy and an `AfterEach` which cleans this up again, with the `It` statement being the one that creates and destroys a MachineSet
```suggestion
	It("create machines when configured behind a proxy", func() {
		By("creating host for proxy")
		client, err := framework.LoadClient()
		Expect(err).NotTo(HaveOccurred())
		proxyMachineSet, err := framework.CreateProxyMachineSet(client)
		Expect(err).NotTo(HaveOccurred())

		By("deploying an HTTP proxy")
		deployment, err := framework.DeployClusterProxy(client)
		Expect(err).NotTo(HaveOccurred())
		Expect(framework.IsDeploymentAvailable(client, deployment.Name, deployment.Namespace)).To(BeTrue())

		By("configuring cluster-wide proxy")
		pods, err := framework.GetPods(client, map[string]string{"app": "squid"})
		Expect(err).NotTo(HaveOccurred())
		Expect(len(pods.Items) > 0).To(BeTrue())
		proxy := configv1.Proxy{}
		proxy.Spec.HTTPProxy = "http://" + pods.Items[0].Status.HostIP + ":80"
		proxy.Spec.HTTPSProxy = "http://" + pods.Items[0].Status.HostIP + ":80"
		framework.SetClusterProxy(client, proxy)
		Expect(err).NotTo(HaveOccurred())

		By("waiting for machine-api-controller deployment to reflect configured cluster-wide proxy")
		Expect(framework.WaitForProxyInjectionSync(client, maoManagedDeployment, framework.MachineAPINamespace, true)).To(BeTrue())

		By("creating a machineset")
		machineSetParams = framework.BuildMachineSetParams(client, 3)
		machineSet, err = framework.CreateMachineSet(client, machineSetParams)
		Expect(err).ToNot(HaveOccurred())
		framework.WaitForMachineSet(client, machineSet.GetName())

		By("destroying a machineset")
		Expect(client.Delete(context.Background(), machineSet)).To(Succeed())
		framework.WaitForMachineSetDelete(client, machineSet)

		By("unconfiguring cluster-wide proxy")
		proxy := configv1.Proxy{}
		proxy.Spec.HTTPProxy = ""
		proxy.Spec.HTTPSProxy = ""
		framework.SetClusterProxy(client, proxy)
		Expect(err).NotTo(HaveOccurred())

		By("waiting for machine-api-controller deployment to reflect unconfigured cluster-wide proxy")
		Expect(framework.WaitForProxyInjectionSync(client, maoManagedDeployment, framework.MachineAPINamespace, false)).To(BeTrue())
		framework.DestroyClusterProxy(client)

		Expect(proxyMachineSet != nil).To(BeTrue())
		framework.DeleteMachineSets(client, proxyMachineSet)
	})
```